app/main.py

```diff
@@ -1,2 +1,66 @@
+from typing import Any, Union
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_size = 8
+        self.hash_table = [[] for _ in range(self.hash_size)]
+        self.threshold = int(self.hash_size * 2 / 3)
+
+    def resize_hash(self) -> None:
+        old_hash_table = self.hash_table
+        self.hash_size *= 2
+        self.threshold = int(self.hash_size * 2 / 3)
+        self.length = 0
+        self.hash_table = [[] for _ in range(self.hash_size)]
+        for item in old_hash_table:
+            if item:
+                self.__setitem__(item[0], item[2])
```

Comment:
This will affect changing old_hash_table. The list is mutable and passed by href. You must copy or deepcopy it